src/status_im/protocol/web3/inbox.cljs

```diff
@@ -19,13 +20,16 @@
       {:error (.-message e)})))
 
 (defn- response-handler [error-fn success-fn]
-  (fn [response]
-    (let [{:keys [error result]} (parse-json response)]
-      ;; NOTE(dmitryn): AddPeer() may return {"error": ""}
-      ;; assuming empty error is a success response
-      (if (seq error)
-        (error-fn error)
-        (success-fn result)))))
+  (fn handle-response
+    ([response]
+     (let [{:keys [error result]} (parse-json response)]
+       (handle-response error result)))
+    ([error result]
```

Comment:
What is the format of `error` and `result` args? 
`handle-response` fn is assumed to call callback fn (error-fn or success-fn) with clj data structure, not JSON (see `js-clj` conversion in `parse-json`)

Reply:
@dmitryn `error` could be `js/Error` object when you call from web3.js API. It would be `nil` otherwise. And `result` is just `true` when the calback is success from the API calls `markTrustedPeer` and `requestMessages`.
Not sure if I should do the conversion in `parse-json` when directly calling from web3.js API here.